tests/integration/reqs/test_subscribe.py

```diff
@@ -0,0 +1,94 @@
+from tests.integration.integration_test_case import IntegrationTestCase
+from tests.integration.it_utils import test_async_and_sync
+from xrpl.models.requests import StreamParameter, Subscribe
+from xrpl.models.requests.unsubscribe import Unsubscribe
+
+_MESSAGE_LIMIT = 3
+
+
+class TestSubscribe(IntegrationTestCase):
+    @test_async_and_sync(globals(), websockets_only=True)
+    async def test_ledger_subscription(self, client):
+        await client.send(Subscribe(streams=[StreamParameter.LEDGER]))
+        count = 0
+        async for message in client:
+            if count == 0:
+                self.assertIsInstance(message["result"]["ledger_index"], int)
+            else:
+                self.assertIsInstance(message["ledger_index"], int)
+            if count == _MESSAGE_LIMIT:
+                break
+            count += 1
+
+        await client.send(Unsubscribe(streams=[StreamParameter.LEDGER]))
```

Comment:
you could use `await client.request(Unsubscribe(streams=[StreamParameter.LEDGER]))` to avoid another `for` loop...i believe

Reply:
if i'm right - let's replace the unsubscribe parts with this pattern since it's more terse

Reply:
I still need a loop to check if the messages contain a successful unsubscription response.